routing/traffic_stash.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "routing/num_mwm_id.hpp"
+
+#include "traffic/traffic_cache.hpp"
+#include "traffic/traffic_info.hpp"
+
+#include "indexer/mwm_set.hpp"
+
+#include <unordered_map>
+
+namespace routing
+{
+class TrafficStash final
+{
+public:
+  TrafficStash(traffic::TrafficCache const & source) : m_source(source) {}
+
+  traffic::TrafficInfo::Coloring const * Get(NumMwmId numMwmId) const
+  {
+    auto it = m_mwmToTraffic.find(numMwmId);
+    if (it == m_mwmToTraffic.cend())
+      return nullptr;
+
+    return it->second.get();
+  }
+
+  void Load(MwmSet::MwmId const & mwmId, NumMwmId numMwmId)
+  {
+    auto traffic = m_source.GetTrafficInfo(mwmId);
+    if (traffic)
+      m_mwmToTraffic[numMwmId] = traffic;
+  }
+
+  class Guard final
```

Comment:
All internal classes should be declared first (https://google.github.io/styleguide/cppguide.html#Declaration_Order)